components/browser/session/src/test/java/mozilla/components/browser/session/SessionTest.kt

```diff
@@ -91,6 +94,19 @@ class SessionTest {
         verifyNoMoreInteractions(observer)
     }
 
+    @Test
+    fun `action is dispatched when URL changes`() {
+        val store: BrowserStore = mock()
+        `when`(store.dispatch(any())).thenReturn(mock())
+
+        val session = Session("https://www.mozilla.org")
+        session.store = store
+        session.url = "http://www.firefox.com"
+
+        verify(store).dispatch(ContentAction.UpdateUrlAction(session.id, session.url))
+        verifyNoMoreInteractions(store)
+    }
```

Comment:
Personally I'd prefer more of a blackbox test here that verifies that an actual `Session`<-> `(Custom)TabSessionState` pair is kept in sync - instead of verifying an implementation detail (which action is dispatched). But then again this test is good to have too.